frontend/app/(defaultLayout)/products/app-router/[handle]/json-ld-scripts.tsx

```diff
@@ -0,0 +1,126 @@
+   BreadcrumbList as BreadcrumbListLDSchema,
+   OfferItemCondition,
+   Product as ProductLDSchema,
+} from 'schema-dts';
+
+interface ProductBreadCrumbsJsonLDScriptProps {
+   product: Product;
+}
+
+export function ProductBreadcrumbsJsonLDScript({
+   product,
+}: ProductBreadCrumbsJsonLDScriptProps) {
+   return (
+      <script
+         {...jsonLdScriptProps<BreadcrumbListLDSchema>({
+            '@context': 'https://schema.org',
+            '@type': 'BreadcrumbList',
+            itemListElement:
+               product.breadcrumbs.map((item, index) => ({
+                  '@type': 'ListItem',
+                  position: index + 1,
+                  name: item.label,
+                  item:
+                     item.url && item.url !== '#'
+                        ? `${IFIXIT_ORIGIN}${item.url}`
+                        : undefined,
+               })) || undefined,
+         })}
+      />
+   );
+}
+
+interface ProductJsonLDScriptProps {
+   product: Product;
+   selectedVariantId: string;
+}
+
+export function ProductJsonLDScript({
+   product,
+   selectedVariantId,
+}: ProductJsonLDScriptProps) {
+   const variant = variantFor(selectedVariantId);
+
+   if (variant == null) return null;
+
+   return (
+      <script
+         {...jsonLdScriptProps<ProductLDSchema>({
+            '@context': 'https://schema.org',
+            '@type': 'Product',
+            name: product.metaTitle || undefined,
+            url: variantUrl(),
+            aggregateRating: shouldShowProductRating(product.reviews)
+               ? {
+                    '@type': 'AggregateRating',
+                    ratingValue: product.reviews.rating,
+                    reviewCount: product.reviews.count,
+                 }
+               : undefined,
+            brand: {
+               '@type': 'Brand',
+               name: product.vendor || 'iFixit',
+            },
+            description: product.shortDescription || '',
+            image: imagesFor(product, selectedVariantId).map(
+               (image) => image.url
+            ),
+            mpn: variant.sku || undefined,
+            offers: {
+               '@type': 'Offer',
+               url: variantUrl(),
+               priceCurrency: variant.price.currencyCode,
+               price: variant.price.amount,
+               priceValidUntil: oneYearFromNow(),
+               itemCondition: variantCondition(),
+               availability:
+                  variant.quantityAvailable && variant.quantityAvailable > 0
+                     ? 'https://schema.org/InStock'
+                     : 'https://schema.org/OutOfStock',
+            },
+            sku: parseItemcode(variant.sku || '').productcode,
+         })}
+      />
+   );
+
+   function variantFor(variantId: string) {
+      return product.variants.find((variant) => variant.id === variantId);
+   }
+
+   function variantUrl() {
+      return `${IFIXIT_ORIGIN}/products/${product.handle}?variant=${selectedVariantId}`;
+   }
```

Comment:
I'm not sure if this matters, but in the past we've intentionally excluded the `variant` query param from the url when we are dealing with the default variant.

Reply:
You are right! I'm adding a note to fix it before we ship 👍